engine/api/src/org/apache/cloudstack/engine/subsystem/api/storage/PrimaryDataStoreDriver.java

```diff
@@ -25,6 +25,13 @@
 import com.cloud.storage.StoragePool;
 
 public interface PrimaryDataStoreDriver extends DataStoreDriver {
+    String BASIC_CREATE = "basicCreate";
+    String BASIC_DELETE = "basicDelete";
+    String BASIC_DELETE_FAILURE = "basicDeleteFailure";
+    String BASIC_GRANT_ACCESS = "basicGrantAccess";
+    String BASIC_REVOKE_ACCESS = "basicRevokeAccess";
+    String BASIC_IQN = "basicIqn";
```

Comment:
These are like what we did with snapshots and cloning in a previous PR: Ways of passing info into the storage driver to have it perform actions that aren't covered by the current interface.

For example, to make this PR work, we need to have two backend volumes exist for the same CloudStack volume simultaneously (during the copy of the data).

"basicCreate" in volume_details for a given CloudStack volume ID tells the storage driver to simply create a volume on the backend with the same characteristics as the existing volume on the backend (for this CloudStack volume).

We then communicate info back to the storage framework (for XenMotion/vMotion purposes) by adding "basicIqn" to the volume details.